internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -825,11 +895,91 @@ func resourceKubernetesClusterNodePoolUpdate(d *pluginsdk.ResourceData, meta int
 		props.MinCount = nil
 	}
 
-	log.Printf("[DEBUG] Updating existing %s..", *id)
-	existing.Model.Properties = props
```

Comment:
These two lines are very important and need to be added back in since we're collecting all of the changes made to a resource in the `props` object.

The existing resource's `Properties` needs to be replaced/overridden with the object `props` otherwise all references to `existing.Model` won't contain the changes made to the resource.

Reply:
The variable `props` is created above by `props := existing.Model.Properties`, which means that the props variable will hold a reference to the original Properties field of existing.Model. Therefore, any changes made to props will also affect existing.Model.Properties and is not necessary to reassign back. 

The log line has been moved down, in the else part of `if cycleNodePool {`.